src/components/answer-page/answer-page.js

```diff
@@ -0,0 +1,204 @@
+import { connect } from 'react-redux';
+import { updateQuestions, updateOffer } from '../../redux/actions.js';
+import Question from '../question/question';
+import ModalOfferHelp from './../modal/modal-offer-help';
+
+class AnswerPage extends Component {
+
+  state = {
+    questions: null,
+    loggedIn: false,
+    showModal: false,
+    modalRef: {
+      questionid: null,
+      learner: null,
+    },
+    offlineUsers: null,
+    allUsers: null
+  }
+
+  getQuestions = async () => {
+    const token = localStorage.getItem('token');
+    if (token) {
+      this.setState({loggedIn: true})
+    }
+    fetch(`${process.env.REACT_APP_SERVER_URL}/questions`, {
+      method: 'GET',
+      headers : {
+        'Authorization' : 'Bearer ' + token,
+        'Content-Type': 'application/json'
+      }})
+    .then(res => res.json())
+    .then(res => this.setState({
+      questions: res
+    }))
+  }
+
+  getUsers = () => {
+    const token = localStorage.getItem('token');
+    if (token) {
+      fetch(`${process.env.REACT_APP_SERVER_URL}/users/`, {
+        method: 'GET',
+        headers : {
+          'Authorization' : 'Bearer ' + token,
+          'Content-Type': 'application/json'
+        }})
+      .then(res => res.json())
+      .then(res => {
+        this.setState({
+          offlineUsers: res.filter(user => {
+            return user.available === null;
+          })
+        });
+        this.setState({
+          allUsers: res
+        });
+      })
+    }
+  }
+
+  sendOffer = (details) => {
+    const token = localStorage.getItem('token');
+    fetch(`${process.env.REACT_APP_SERVER_URL}/questions/${details.questionid}/offers`, {
+      method: 'POST',
+      headers: {
+        'Authorization' : 'Bearer ' + token,
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        "message": details.message,
+        "expiration": details.expiration
+      })
+    })
+      .then(res => res.json())
+      .then(offer => {
+        this.props.updateOffer(offer);
+        const currentQuestion = this.state.questions.filter(question => question.question_id === offer.linked_question);
+        this.props.socket.emit('offer sent', { offer, learner_id: currentQuestion[0].learner })
+      })
+  }
+
+  showOfferModal = () => {
+    return <ModalOfferHelp questions={this.state.questions} users={this.state.allUsers} showModal={this.state.showModal} modalRef={this.state.modalRef} closeOfferModal={this.closeOfferModal} sendOffer={this.sendOffer}/>
+  }
+
+  openOfferModal = (question) => {
+    this.setState(state => {
+      state.modalRef.questionid = question.question_id
+      state.modalRef.learner = question.learner
+    })
+    this.setState({
+      showModal: true
+    })
+  }
+
+  closeOfferModal = () => {
+    //close it here
+    this.setState({
+      showModal: false
+    })
+  }
+
+  renderQuestions = () => {
+    //User is not logged in
```

Comment:
This function needs a refactor.

Maybe I'm missing something, but after some detailed reading, all this questions processing seems like you're just filtering out any question which status is not included into `['help-now', 'pending', 'closed']` and the learner is online. Which is:

```js
this.state.questions.filter(question => 
  !['help-now', 'pending', 'closed'].includes(question.status)
  || !this.state.offlineUsers.find(user => user.user_id === question.learner)
)
```

Reply:
Anyways, all this processing in the render makes it difficult to read, try some abstractions. For example:

```js
const questionsHelpNow = this.state.questions.filter((question) => question.isHelpNow() && question.learnerIsOffline());
      
```
is more readable than:

```js
const questionsHelpNow = this.state.questions.filter((question) => question.status==='help-now' && this.state.offlineUsers.filter(user => user.user_id===question.learner).length===0);
```